conventional_commits.go

```diff
@@ -0,0 +1,132 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/google/go-github/v28/github"
+	"github.com/sirupsen/logrus"
+
+	clientgithub "github.com/mendersoftware/integration-test-runner/client/github"
+	"github.com/mendersoftware/integration-test-runner/git"
+)
+
+const (
+	footer = `Changelog: All
+Ticket: None`
+	commentErrorPrexix = "I did my very best, but:\n"
+)
+
+func getLegalTypes() []string {
+	return []string{"fix", "feat"}
+}
+
+func conventionalComittifyDependabotPr(
+	log *logrus.Entry,
+	comment *github.IssueCommentEvent,
+	pr *github.PullRequest,
+	conf *config,
+	body string,
+	githubClient clientgithub.Client,
+) error {
+	err := attemptConventionalComittifyDependabotPr(log, pr, body)
+
+	if err == nil {
+		return nil
+	}
+
+	commentBody := commentErrorPrexix + err.Error()
+	if err := githubClient.CreateComment(
+		context.Background(),
+		conf.githubOrganization,
+		comment.GetRepo().GetName(),
+		pr.GetNumber(),
+		&github.IssueComment{Body: &commentBody}); err != nil {
+		log.Infof("Failed to comment on the pr: %v, Error: %s", pr, err.Error())
+	}
+
+	return err
+}
+
+func attemptConventionalComittifyDependabotPr(
+	log *logrus.Entry,
+	pr *github.PullRequest,
+	body string,
+) error {
+
+	typeKeyword, err := getTypeKeyword(body)
+	if err != nil {
+		return err
+	}
+
+	// take message, and conventional committify it
+	headBranch := pr.GetHead().GetRef()
+	cloneUrl := *pr.GetHead().GetRepo().CloneURL
+	state, err := git.Commands(
+		git.Command("clone", "-b", headBranch, "--single-branch", cloneUrl),
+	)
+	defer state.Cleanup()
+
+	if err != nil {
+		log.Infof("Could not clone branch %s from %s", headBranch, cloneUrl)
+		return err
+	}
+
+	messageBytes, err := git.Command("--no-pager", "show", "-s", "--format=%B", "HEAD").
+		With(state).CombinedOutput()
+	if err != nil {
+		log.Info(err.Error())
+		return err
+	}
+
+	message := string(messageBytes)
+
+	newMessage := conventionalComittifyDependabotMessage(message, typeKeyword)
+
+	if err := git.CommandsWithState(
+		state,
+		git.Command("commit", "--amend", "-m", newMessage),
+		git.Command("push", "--force"),
+	); err != nil {
+		log.Info("Could not amend and push")
+		return err
+	}
+
+	return nil
+}
+
+func conventionalComittifyDependabotMessage(message string, typeKeyword string) string {
+	message = strings.TrimSpace(message)
+	fmt.Print(message)
+	message = strings.TrimPrefix(message, "Changelog:All: ")
+	message = typeKeyword + ": " + message
+
+	fi := strings.LastIndex(message, "Signed-off-by")
+	if fi < 0 {
+		message += "\n\n"
+		fi = len(message)
+	}
+	message = message[:fi] + footer + "\n" + message[fi:]
+
+	return strings.TrimSpace(message)
+}
+
+func getTypeKeyword(s string) (string, error) {
```

Comment:
Suggestion:
I think I would extract the wanted type with a regex.

It's a lot easier to read. So just match on the wanted string, and extract the group which surrounds the type.